loopy/target/pyopencl.py

```diff
@@ -370,6 +370,38 @@ def map_power(self, expr, type_context):
 # }}}
 
 
+# {{{ filter out repeated base storages in IDIS
+
+def convert_global_temp_args_to_base_storage(kernel, idis):
```

Comment:
Not sure I love the name. It sure gave me the wrong impression. Based on the function name, I thought this was going to take all global temporaries and stick them into one big buffer. Instead, it seems to destroy information in the IDI?